### PR TITLE
Twisted 15.3.0 does not raises expected exception serializing lambda …

### PR DESCRIPTION
**Pull request by dangra**
Twisted bug is described in https://twistedmatrix.com/trac/ticket/7989

### CODE REVIEW

File: tests/test_squeues.py [hunk 1]
```diff
@@ -31,6 +32,8 @@ def test_serialize(self):
         self.assertEqual(q.pop(), {'a': 'dict'})
 
     def test_nonserializable_object(self):
+        # Trigger Twisted bug #7989
+        import twisted.persisted.styles  # NOQA
```
Comment (eliasdorneles) [inline tests/test_squeues.py]: considering that the module is already imported at the top, won't this line be a no-op?

**Comment by eliasdorneles** [comment]
For some reason, pickling lambda is failing to raise an error only in python 3.3 (even though it still doesn't work properly):

```$ python
Python 3.3.6 (default, Jul 29 2015, 18:58:09) 
[GCC 4.8.4] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pickle
>>> pickle.dumps(lambda x: x)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
_pickle.PicklingError: Can't pickle <class 'function'>: attribute lookup builtins.function failed
>>> import twisted.persisted.styles
>>> pickle.dumps(lambda x: x)
b'\x80\x03ctwisted.persisted.styles\n_unpickleFunction\nq\x00X\x11\x00\x00\x00__main__.<lambda>q\x01\x85q\x02Rq\x03.'
>>> x = pickle.loads(pickle.dumps(lambda x: x))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/elias/.virtualenvs/tmp-6d2d378f3497cd1b/lib/python3.3/site-packages/twisted/persisted/styles.py", line 136, in _unpickleFunction
    return namedAny(fullyQualifiedName)
  File "/home/elias/.virtualenvs/tmp-6d2d378f3497cd1b/lib/python3.3/site-packages/twisted/python/reflect.py", line 314, in namedAny
    obj = getattr(obj, n)
AttributeError: 'module' object has no attribute '<lambda>'
```

File: tests/test_squeues.py (hunk 1 above)
Comment (dangra) [inline tests/test_squeues.py]: my bad, I only intended to import it here.

**Comment by dangra** [comment]
Do not merge, aside of messy commit history I am not sure this is the correct approach. The side effects of twisted.persisted are triggered because twisted.web.server is imported for mock tests, so in normal crawls this module shouldn't be imported (unless it does by an extension).

An alternative is to force the import of twisted.persisted.styles and then clear entries added by twisted to `copy_reg.dispatch_table`.

**Comment by dangra** [comment]
We choose an alternative fix: #1414

**Pull request closed without merging**